dingo/core/result.py

```diff
@@ -276,17 +276,26 @@ def importance_sample(self, num_processes: int = 1, **likelihood_kwargs):
         valid_samples = (log_prior + delta_log_prob_target) != -np.inf
         theta = theta.iloc[valid_samples]
 
-        print(f"Calculating {len(theta)} likelihoods.")
-        t0 = time.time()
-        log_likelihood = self.likelihood.log_likelihood_multi(
-            theta, num_processes=num_processes
-        )
-        print(f"Done. This took {time.time() - t0:.2f} seconds.")
+        if "log_likelihood" not in self.samples.columns:
+            print(f"Calculating {len(theta)} likelihoods.")
+            t0 = time.time()
+            log_likelihood = self.likelihood.log_likelihood_multi(
+                theta, num_processes=num_processes
+            )
+            print(f"Done. This took {time.time() - t0:.2f} seconds.")
+        else:
+            log_likelihood = self.samples["log_likelihood"]
+            log_likelihood = log_likelihood[valid_samples]
```

Comment:
Why is it necessary to add this `else` statement? Importance sampling should be calculating the likelihood itself.